### PR TITLE
Deprecate the collectd/health-checker plugin

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Deprecate the collectd/health-checker plugin.

### CODE REVIEW

**Review by crobert-1 — APPROVED** [review]
Thanks for including the replacement in the warning message 👍

File: internal/signalfx-agent/pkg/monitors/collectd/healthchecker/healthchecker.go [hunk 1]
```diff
@@ -104,6 +104,6 @@ func (m *Monitor) Configure(conf *Config) error {
 	if conf.Name == "" {
 		conf.pyConf.PluginConfig["Instance"] = fmt.Sprintf("%s-%d", conf.Host, conf.Port)
 	}
-
+	m.Logger().Warn("[NOTICE] The collectd/health-checker plugin is deprecated and will be removed in a future release. Use the httpcheckreceiver instead.")
```
Comment (aunshc) [inline internal/signalfx-agent/pkg/monitors/collectd/healthchecker/healthchecker.go:107]:
```suggestion
	m.Logger().Warn("[NOTICE] The collectd/health-checker plugin is deprecated and will be removed in a future release. Use the opentelemetry httpcheck receiver instead.")
```

Comment (crobert-1) [reply internal/signalfx-agent/pkg/monitors/collectd/healthchecker/healthchecker.go:107]: Nit on the suggestion: We'd probably want it to be `opentelemetry` -> `OpenTelemetry`

File: internal/signalfx-agent/pkg/monitors/collectd/healthchecker/metadata.yaml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 monitors:
 - dimensions:
   doc: |
+    The collectd/health-checker plugin is deprecated and will be removed in a future release. Use the httpcheckreceiver instead.
```
Comment (aunshc) [inline internal/signalfx-agent/pkg/monitors/collectd/healthchecker/metadata.yaml:4]:
```suggestion
    The collectd/health-checker plugin is deprecated and will be removed in a future release. Use the [OpenTelemetry HTTP Check Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/httpcheckreceiver) instead.
```

Minor update for consistency

**Pull request merged**